### PR TITLE
Allow federation to send additional config values to clients

### LINKED ISSUE #1432: Allow additional fields to be transmitted to clients in config

### PR DESCRIPTION
**Pull request by elsirion**
Fixes #1432

### CODE REVIEW

File: client/cli/src/main.rs [hunk 1]
```diff
@@ -542,9 +542,24 @@ async fn handle_command(
                 .iter()
                 .map(|(amount, coins)| (amount.to_owned(), coins.len()))
                 .collect();
+
+            let federation_name = client
+                .config()
+                .as_ref()
+                .meta
+                .get("federation_name")
```
Comment (jkitman) [inline client/cli/src/main.rs:550]: Can we make this a `const`?

Comment (elsirion) [reply client/cli/src/main.rs]: Done.

File: client/cli/src/main.rs [hunk 2]
```diff
@@ -542,9 +542,24 @@ async fn handle_command(
                 .iter()
                 .map(|(amount, notes)| (amount.to_owned(), notes.len()))
                 .collect();
+
+            let federation_name = client
+                .config()
+                .as_ref()
+                .meta
+                .get(META_FEDERATION_NAME_KEY)
+                .ok_or_else(|| {
```
Comment (dpc) [inline client/cli/src/main.rs:551]: If it's required... why make it an "additional config value"?

Comment (elsirion) [reply client/cli/src/main.rs:551]: I think it was a suggestion by @justinmoon to make it not required anymore. There could be a BIP-like process to define meta-data extensions. And even though most federations will have a name, it's not *really* necessary.

Comment (dpc) [reply client/cli/src/main.rs:551]: Makes sense, but then we should remove this error here? Or default to "unnamed" for now?

Comment (elsirion) [reply client/cli/src/main.rs:551]: Right, the client needs to be able to just display all the meta fields.

**Comment by elsirion** [comment]
One limitation is that it only allows `String -> String` maps, so more complex data structures would need some additional encoding. This is due to the need to hash the config for signing, so if we wanted more general data types we'd need to create a `DynMetaEntry` struct with the `trait IMetaEntry: Serialize + Deserialize + Encodable` so that it's both compatible with `serde` and our own encoding API.

File: client/cli/src/main.rs (hunk 2 above)
Comment (elsirion) [reply client/cli/src/main.rs]: Done.

File: fedimint-api/src/config.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (justinmoon) [inline fedimint-api/src/config.rs:1]: What it the purpose of this empty file?

**Review by justinmoon — APPROVED** [review]

File: fedimint-api/src/config.rs (hunk 3 above)
Comment (elsirion) [reply fedimint-api/src/config.rs]: Good catch, rebasing artifact.

**Review by justinmoon — APPROVED** [review]

**Review by dpc — APPROVED** [review]

**Pull request merged**